parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -119,7 +120,12 @@ private JSONArray getTeachers() throws IOException, JSONException {
 	}
 
 	private JSONArray getJSONArray(String url) throws IOException, JSONException {
-		return new JSONArray(httpGet(url, "UTF-8"));
+		try {
+			return new JSONArray(httpGet(url, "UTF-8"));
+		} catch (HttpResponseException e) {
+			e.printStackTrace();
```

Comment:
Ah, okay, die API gibt `404` zurück wenn keine Lehrer bzw. Kurse existieren. Dann würde ich die Exception erst in getAllClasses und getAllTeachers fangen, aber dann nicht `e.printStackTrace()` sondern prüfen, was der Fehlercode ist und bei `404` direkt `return null` und sonst mit `throw e` die Exception weiterreichen (bzw. letztendlich anhalten).